File: Matury CKE/2021 Maj/Programowanie/C++/zad4_4.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main()
{
    ifstream dane("C:/Matury-Informatyka/Matury CKE/2021 Maj/DANE_2105/instrukcje.txt");
    // ifstream dane("C:/Matury-Informatyka/Matury CKE/2021 Maj/DANE_2105/przyklad.txt");

    string odpowiedz = "";

    string instrukcja; char argument;
    while(dane >> instrukcja >> argument){
        if(instrukcja == "DOPISZ"){
            odpowiedz += argument;
        }else if(instrukcja == "USUN"){
            odpowiedz.pop_back();
        }else if(instrukcja == "ZMIEN"){
            odpowiedz.pop_back();
            odpowiedz += argument;
        }else if(instrukcja == "PRZESUN"){
            int idx = odpowiedz.find(argument);
            if(!idx && idx != 0)continue;
            int i = (int)argument;
            char a = (i==90 ? 65 : i+1);
            odpowiedz[idx] = a;
        }
    }

    cout << "4.4. " << odpowiedz;

    return 0;
}

File: Matury CKE/2021 Maj/Programowanie/Python/zad4_4.py
import string

t = []
alfabet = list(string.ascii_uppercase)  # ABCDEFGHIJKLMNOPQRSTUVWXYZ

with open("../../DANE_2105/instrukcje.txt", "r") as plik:
    for linia in plik.readlines():
        cmd, litera = linia.strip().split(" ")
        if cmd == "DOPISZ":
            t.append(litera)
        elif cmd == "ZMIEN":
            t[-1] = litera
        elif cmd == "USUN":
            t.pop(-1)
        elif cmd == "PRZESUN":
            nast_litera = 'A' if litera == 'Z' else alfabet[alfabet.index(litera)+1]
            t[t.index(litera)] = nast_litera

print(''.join(t))
